scripts/performance-check.sh

```diff
@@ -1,44 +1,110 @@
+for i in `seq 1 $WARMUP_COUNT`; do
+    $ROOT_DIR/scripts/gear.sh test gear --release > $ROOT_DIR/target/tests-output/$i 2>&1
+done
+
 for i in `seq 1 $COUNT`; do
-    $ROOT_DIR/scripts/gear.sh test gear --release
+    echo $i
+    $ROOT_DIR/scripts/gear.sh test gear --release > $ROOT_DIR/target/tests-output/$i 2>&1
     mv $ROOT_DIR/target/nextest/ci/junit.xml $ROOT_DIR/target/tests/$i
 done
 
-mkdir $ROOT_DIR/target/runtime-tests/
+mkdir -p $ROOT_DIR/target/runtime-tests/
+mkdir -p $ROOT_DIR/target/runtime-tests-output/
+for i in `seq 1 $WARMUP_COUNT`; do
+    $ROOT_DIR/scripts/gear.sh test rtest > $ROOT_DIR/target/runtime-tests-output/$i 2>&1
+done
+
 for i in `seq 1 $COUNT`; do
-    $ROOT_DIR/scripts/gear.sh test rtest
+    echo $i
+    $ROOT_DIR/scripts/gear.sh test rtest > $ROOT_DIR/target/runtime-tests-output/$i 2>&1
     mv $ROOT_DIR/target/runtime-test-junit.xml $ROOT_DIR/target/runtime-tests/$i
 done
 
-mkdir $ROOT_DIR/target/main_branch/
+rm -rf $ROOT_DIR/target/main_branch/
+mkdir -p $ROOT_DIR/target/main_branch/
 cargo run --package regression-analysis --release -- collect-data --data-folder-path $ROOT_DIR/target/tests/ --output-path $ROOT_DIR/target/main_branch/pallet-tests.json
 cargo run --package regression-analysis --release -- collect-data --disable-filter --data-folder-path $ROOT_DIR/target/runtime-tests/ --output-path $ROOT_DIR/target/main_branch/runtime-tests.json
 
 git checkout $CURRENT_BRANCH
 
-$ROOT_DIR/scripts/gear.sh test gear --release
-$ROOT_DIR/scripts/gear.sh test rtest
+make gear-release
+make examples
+
+rm -rf $ROOT_DIR/target/tests/*
+for i in `seq 1 $WARMUP_COUNT`; do
+    $ROOT_DIR/scripts/gear.sh test gear --release > $ROOT_DIR/target/tests-output/$i 2>&1
+done
+
+for i in `seq 1 $COUNT`; do
+    echo $i
+    $ROOT_DIR/scripts/gear.sh test gear --release > $ROOT_DIR/target/tests-output/$i 2>&1
+    mv $ROOT_DIR/target/nextest/ci/junit.xml $ROOT_DIR/target/tests/$i
+done
+
+rm -rf $ROOT_DIR/target/runtime-tests/*
+for i in `seq 1 $WARMUP_COUNT`; do
+    $ROOT_DIR/scripts/gear.sh test rtest > $ROOT_DIR/target/runtime-tests-output/$i 2>&1
+done
+
+for i in `seq 1 $COUNT`; do
+    echo $i
+    $ROOT_DIR/scripts/gear.sh test rtest > $ROOT_DIR/target/runtime-tests-output/$i 2>&1
+    mv $ROOT_DIR/target/runtime-test-junit.xml $ROOT_DIR/target/runtime-tests/$i
+done
+
+rm -rf $ROOT_DIR/target/current_branch/
+mkdir -p $ROOT_DIR/target/current_branch/
+cargo run --package regression-analysis --release -- collect-data --data-folder-path $ROOT_DIR/target/tests/ --output-path $ROOT_DIR/target/current_branch/pallet-tests.json
+cargo run --package regression-analysis --release -- collect-data --disable-filter --data-folder-path $ROOT_DIR/target/runtime-tests/ --output-path $ROOT_DIR/target/current_branch/runtime-tests.json
+
+if [ "$BOXPLOT" != "0" ]; then
+    python3 $ROOT_DIR/scripts/performance-boxplot.py $ROOT_DIR/target/main_branch/pallet-tests.json $ROOT_DIR/target/current_branch/pallet-tests.json
```

Comment:
and provide ability to just run box plot on pre-generated results

Reply:
I would like not to add additional logic to this wrapper script to keep it simple. In any case one can run this python-script by hand on pre-generated data.